.github/workflows/build.yml

```diff
@@ -349,37 +353,60 @@ jobs:
+              "./$EXTRACTED/aarch64-apple-ios-sim/libradix_engine_toolkit_uniffi.dylib" \
+              "./$EXTRACTED/x86_64-apple-ios/libradix_engine_toolkit_uniffi.dylib" \
+              -o "./$EXTRACTED/ios-simulator-arm64_x86_64/libradix_engine_toolkit_uniffi.dylib"
+
+          # Set proper id, so that Xcode can properly locate the dynamic library
+          install_name_tool -id @rpath/libradix_engine_toolkit_uniffi.dylib ./$EXTRACTED/aarch64-apple-ios/libradix_engine_toolkit_uniffi.dylib
+          install_name_tool -id @rpath/libradix_engine_toolkit_uniffi.dylib ./$EXTRACTED/macos-arm64_x86_64/libradix_engine_toolkit_uniffi.dylib
+          install_name_tool -id @rpath/libradix_engine_toolkit_uniffi.dylib ./$EXTRACTED/ios-simulator-arm64_x86_64/libradix_engine_toolkit_uniffi.dylib
 
           # Creating the XCFramework
           xcodebuild -create-xcframework \
-              -library "./$EXTRACTED/aarch64-apple-ios/libradix_engine_toolkit_uniffi.a" \
+              -library "./$EXTRACTED/aarch64-apple-ios/libradix_engine_toolkit_uniffi.dylib" \
               -headers "./$EXTRACTED/include" \
-              -library "./$EXTRACTED/macos-arm64_x86_64/libradix_engine_toolkit_uniffi.a" \
+              -library "./$EXTRACTED/macos-arm64_x86_64/libradix_engine_toolkit_uniffi.dylib" \
               -headers "./$EXTRACTED/include" \
-              -library "./$EXTRACTED/ios-simulator-arm64_x86_64/libradix_engine_toolkit_uniffi.a" \
+              -library "./$EXTRACTED/ios-simulator-arm64_x86_64/libradix_engine_toolkit_uniffi.dylib" \
               -headers "./$EXTRACTED/include" \
               -output "./$EXTRACTED/RadixEngineToolkitUniFFI.xcframework"
 
-          # Push changes to Swift Engine Toolkit
-          SWIFT_ENGINE_TOOLKIT="https://github.com/radixdlt/swift-engine-toolkit"
-          git clone $SWIFT_ENGINE_TOOLKIT
-          git checkout $BRANCH || git checkout -b $BRANCH
+      - name: Publish to Swift Engine Toolkit
+        run: |
+          BRANCH=${{ github.ref_name }}
+          BUILD_CRATE_MANIFEST="./radix-engine-toolkit-uniffi/Cargo.toml"
+          CRATE_VERSION=$(cargo read-manifest --manifest-path $BUILD_CRATE_MANIFEST |  jq -r '.version')
+          LAST_COMMIT_SHA=$(git rev-parse --short HEAD)
+          SPM_VERSION="$CRATE_VERSION-$LAST_COMMIT_SHA"
+          SET_UPSTREAM_FLAG=""
 
           cd swift-engine-toolkit
-          git checkout -b $BRANCH_NAME
-          cp -R ../$EXTRACTED/RadixEngineToolkitUniFFI.xcframework ./Sources/RadixEngineToolkit/RadixEngineToolkitUniFFI.xcframework
-          cp -R ../$EXTRACTED/UniFFI-Bindings/output/radix_engine_toolkit_uniffi.swift ./Sources/EngineToolkit/radix_engine_toolkit_uniffi
-
-          LAST_COMMIT_SHA=$(git rev-parse --short HEAD)
-          git commit -am "Update RET from $LAST_COMMIT_SHA"
-          git push origin -b $BRANCH_NAME
+          REMOTE_BRANCH=$(git ls-remote --heads origin $BRANCH | grep $BRANCH)
+
+          if [ -z "$REMOTE_BRANCH" ]
```

Comment:
Just curious, when is this true and when is it false?

Reply:
`true` if $REMOTE_BRANCH  variable is empty, false otherwise